### ISSUE TITLE
Contrast from posterior estimates

### ISSUE THREAD

**Issue by wdonald1985** [richarddmorey/BayesFactor#68]
Although I find the Bayes factor interesting, I am most interested in uncertainty and effect size. When using the lmBF function for the following code: bayesBF <- lmBF(len ~ dose, iterations = 1000, data = myData), the bayes factor does not tell me which groups differ. When using the posterior TRUE option, I notice that I have mu and estimates for all of the levels of dose. Can I add the posterior estimates for dose to mu to get the mcmc mean for each level of dose. Then can I subtract these means, for example, dose.5 - dose.1 to get the contrasts?

**Comment by richarddmorey** [comment]
>   Can I add the posterior estimates for dose to mu to get the mcmc mean for each level of dose. Then can I subtract these means, for example, dose.5 - dose.1 to get the contrasts?

Yes, you can do this. Note though: if you just care about the contrasts, you don't need to add the mean back in, because if you're going to just subtract them then the mean just subtracts right back out. You just need the difference between the effects themselves.

**Comment by wdonald1985** [comment]
When using the random option in lmBF, can I also use the same approach for interactions?
Thank you for the quick reply!

**Comment by richarddmorey** [comment]
I'm not sure what you mean.

**Comment by wdonald1985** [comment]
When there is an interaction, can I subtract posterior estimates of the interactions to determine the difference between interactions.

**Comment by richarddmorey** [comment]
You mean, eg, if there is a three-way interaction, can you take the (say) two two-way interactions and subtract them?

**Comment by wdonald1985** [comment]
Yes. As an aside, I will be starting graduate school this fall at UC Davis where I will be focusing on the neuroendocrinology of social bonding in prarie voles and titi monkeys. As an undergraduate, I completed a minor in statistics, all NHST, but am attempting to train myself in Bayesian methods. 
In my field, journals often ask for effect sizes such as Cohen's d. As such, I am trying to use RSTAN or use a package such as yours for models where I get mean estimates and estimates of sigma for all variables or levels of a factor.
In lmbf, for example, I was considereding obtaining estimates of mu, subtraction to get mean differences, and dividing by the empirical pooled sd. Or just placing a prior that would shrink estimates toward zero and conduct several t.testbf's.

I guess my main concern is that, although I have been reading extensively about Bayesian statistics  (DBDA and Rethinking), my adviser and most people in my field have not. As such, ensuring what I am doing is correct is very, very important.



Thank you for the responses and I am sure that I will figure it all out soon!

**Comment by richarddmorey** [comment]
The three-way interaction parameters itself contain the information, don't they? Unless I'm misunderstanding what you want.
